File: AlgorithmsPackJavaSolutions/Sort an Array.java
public class Solution {
    public int[] sortArray(int[] nums) {
        final int SHIFT = 50000;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            nums[i] += SHIFT;
        }
        
        int maxValue = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] > maxValue) {
                maxValue = nums[i];
            }
        }
        
        int[] count = new int[maxValue + 1];
        for (int i = 0; i < n; i++) {
            count[nums[i]]++;
        }
        
        int idx = 0;
        for (int i = 0; i <= maxValue; i++) {
            for (int j = 0; j < count[i]; j++, idx++) {
                nums[idx] = i - SHIFT;
            }
        }
        
        return nums;
    }
}

File: AlgorithmsPackPythonSolutions/Sort an Array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        SHIFT = 50000
        n = len(nums)
        for i in range(n):
            nums[i] += SHIFT
        
        maxValue = nums[0]
        for i in range(1, n):
            if nums[i] > maxValue:
                maxValue = nums[i]
        
        count = [0] * (maxValue + 1)
        for i in range(n):
            count[nums[i]] += 1
        
        idx = 0
        for i in range(maxValue + 1):
            for j in range(count[i]):
                nums[idx] = i - SHIFT
                idx += 1
        
        return nums
